cmd/uidp/client/client.go

```diff
@@ -433,50 +432,51 @@ func delete(ctx context.Context, rdb *redis.ClusterClient, key string) error {
 
 func hasPermission(ctx context.Context, rdb *redis.ClusterClient, uid, pwd string, newUser User, isCreation bool) error {
 	l := logger.With(zap.String("subSection", "permission"), zap.String("operation", "check"), zap.String("user", uid))
-	requestor, err := get(ctx, rdb, uid)
+	requester, err := get(ctx, rdb, uid)
 	if err != nil {
 		l.Error("error getting user", zap.Error(err))
 		return err
 	}
 
-	if err := bcrypt.CompareHashAndPassword([]byte(requestor.Password), []byte(pwd)); err != nil {
+	if err := bcrypt.CompareHashAndPassword([]byte(requester.Password), []byte(pwd)); err != nil {
 		l.Debug("invalid password")
 		return fmt.Errorf("invalid password for user %v", uid)
 	}
 
-	for newUserPermissionScope, newUserPermissions := range newUser.Permissions {
+	for newUserPermItem, newUserPermScopes := range newUser.Permissions {
 		isAllowed := false
-		for requestorPermissionScope, requestorPermissions := range requestor.Permissions {
-			if isPermissionAllowed(newUserPermissionScope, requestorPermissionScope, newUserPermissions, requestorPermissions, isCreation) {
-				isAllowed = true
-				break
+		reqPermScopes, okPermItem := requester.Permissions[newUserPermItem]
+		_, okToken := requester.Permissions[auth.CreateToken]
+		if !isCreation && okToken {
+			isAllowed = true
+		} else if isCreation && okToken && okPermItem {
+			isAllowed = true
+			for _, newUserScope := range newUserPermScopes {
+				if !isScopeAllowed(newUserScope, reqPermScopes) {
+					isAllowed = false
+					break
+				}
 			}
 		}
-
 		if !isAllowed {
 			l.Debug("user unauthorized")
 			return ierrors.New(
 				"not allowed to create/delete/update a user with current permissions",
 			).Forbidden()
 		}
-
 	}
 
 	return nil
 }
```

Comment:
Not necessary since i will give the approval but would be a nice addition to create a struct that contains the parameters used by the func. 

What i mean is that out of these -> `(ctx context.Context, rdb *redis.ClusterClient, uid, pwd string, newUser User, isCreation bool)` the four elements `uid, pwd, newUser, isCreation` could a part of struct that would be given to the func.

Since is seems long enough to the point that such structure would be welcome for readability.

Also the usage of `isCreation` leads to confusion, considering that implies that it might be a newUser or a existant user, while at the same time you go through the newUser.Permissions without checking. Wouldn't a request made by an old user try to set the `newUser` as `nil`? The way it currently is it is mandatory to pass a structure of User even if its not going to be used.

While i do notice that this was the previous way in which the function was being used, it might be worth to create a more flexible method